railgun/internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -23,15 +25,83 @@ import (
 
 // NewCacheBasedProxy will provide a new Proxy object. Note that this starts some background services
 // and the caller is thereafter responsible for closing the Proxy.StopCh.
-func NewCacheBasedProxy(ctx context.Context, logger logrus.FieldLogger, k8s client.Client, kongConfig sendconfig.Kong, ingressClassName string, processClasslessIngressV1Beta1 bool, processClasslessIngressV1 bool, processClasslessKongConsumer bool) Proxy {
-	return NewCacheBasedProxyWithStagger(ctx, logger, k8s, kongConfig, ingressClassName, processClasslessIngressV1Beta1, processClasslessIngressV1, processClasslessKongConsumer, DefaultStagger)
+func NewCacheBasedProxy(ctx context.Context,
+	logger logrus.FieldLogger,
+	k8s client.Client,
+	kongConfig sendconfig.Kong,
+	ingressClassName string,
+	processClasslessIngressV1Beta1, processClasslessIngressV1, processClasslessKongConsumer, enableReverseSync bool,
+) (Proxy, error) {
+	return NewCacheBasedProxyWithStagger(ctx, logger, k8s, kongConfig, ingressClassName, processClasslessIngressV1Beta1, processClasslessIngressV1, processClasslessKongConsumer, enableReverseSync, DefaultStagger)
 }
 
-func NewCacheBasedProxyWithStagger(ctx context.Context, logger logrus.FieldLogger, k8s client.Client, kongConfig sendconfig.Kong, ingressClassName string, processClasslessIngressV1Beta1 bool, processClasslessIngressV1 bool, processClasslessKongConsumer bool, stagger time.Duration) Proxy {
+func NewCacheBasedProxyWithStagger(ctx context.Context,
+	logger logrus.FieldLogger,
+	k8s client.Client,
+	kongConfig sendconfig.Kong,
+	ingressClassName string,
+	processClasslessIngressV1Beta1, processClasslessIngressV1, processClasslessKongConsumer, enableReverseSync bool,
+	stagger time.Duration,
+) (Proxy, error) {
+	// download the kong root configuration (and validate connectivity to the proxy API)
+	root, err := kongConfig.Client.Root(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// pull the proxy configuration out of the root config and validate it
+	proxyConfig, ok := root["configuration"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("invalid database configuration, expected a string got %t", proxyConfig["database"])
+	}
+
+	// validate the database configuration for the proxy
+	dbmode, ok := proxyConfig["database"].(string)
+	if !ok {
+		return nil, fmt.Errorf("invalid database configuration, expected a string got %t", proxyConfig["database"])
+	}
+	if dbmode == "off" || dbmode == "" {
+		kongConfig.InMemory = true
+	}
```

Comment:
duplicate of what the `switch` below does?

Reply:
yes, removed in b509a67e40aa88e8748ee5ffa5c5288a44516bf5 :+1: